### PR TITLE
Capture the exit code of the test run

### PR DESCRIPTION
**Pull request by cawallin**

### CODE REVIEW

**Comment by cawallin** [comment]
I verified that it successfully worked locally, both if product tests succeeded and failed.

File: ensure_command_output.sh [hunk 1]
```diff
@@ -1,59 +1,43 @@
...
-  PIPE=$1
-  shift
   CMD="$@"
 
-  mkfifo $PIPE
-  echo "Starting command '$CMD' using pipe $PIPE"
-  run_command $PIPE $CMD &
-  run_ping $PIPE $CMD &
+  echo "Starting command '$CMD'"
+  run_command $CMD &
```
Comment (arhimondr) [inline ensure_command_output.sh]: Do you really need to wrap it with a method? Sending bash function to background causes bash fork. And you will get `pid` not of the command itself, but of the bash wrapper. I would rather un-wrap it.

File: ensure_command_output.sh [hunk 2]
```diff
@@ -1,59 +1,43 @@
...
-  (
     while [ 1 ]
     do
       echo "Command '$CMD' is still running."
       sleep $OUTPUT_SLEEP
     done
-  ) >$PIPE
 }
 
 function presto_wait {
```
Comment (arhimondr) [inline ensure_command_output.sh]: What this method has in common with presto? Maybe it is better to express the context of this method within it's name?

**Review by arhimondr — COMMENTED** [review]
I'm not sure i understand the context of this change. It is better to ask someone else to review.

File: ensure_command_output.sh (hunk 1 above)
Comment (cawallin) [reply ensure_command_output.sh]: Yeah, I can do that

File: ensure_command_output.sh [hunk 3]
```diff
@@ -1,59 +1,43 @@
...
 function run_command {
-  PIPE=$1
-  shift
   CMD="$@"
-  $CMD 1>$PIPE 2>$PIPE
-  echo "$MAGIC_END_STRING" 1>$PIPE
+  $CMD 
 }
 
 function run_ping {
```
Comment (arhimondr) [inline ensure_command_output.sh]: I would rather call it `print_with_interval`, and accept two parameters. `message` and `interval`

File: ensure_command_output.sh (hunk 2 above)
Comment (arhimondr) [reply ensure_command_output.sh]: What this method does - it executes the command and produces some dummy output while the command is being executed. I cant make up a good name for it, but it feels that it should be called like `execute_long_running_command_with_output_updates` =)

File: ensure_command_output.sh [hunk 4]
```diff
@@ -1,59 +1,43 @@
...
-      rm -f $PIPE
-      kill -15 $ping_pid
-      exit 0
-    fi
-  done < $PIPE
+ 
+  wait $command_pid
+  exit_code=$?
+      
+  kill -15 $ping_pid
```
Comment (arhimondr) [inline ensure_command_output.sh]: `kill -TERM $ping_pid` ?

File: ensure_command_output.sh [hunk 5]
```diff
@@ -1,59 +1,43 @@
...
+  exit $exit_code
```
Comment (arhimondr) [inline ensure_command_output.sh]: `return $exit_code`

File: ensure_command_output.sh [hunk 6]
```diff
@@ -1,59 +1,43 @@
 #!/bin/bash
```
Comment (arhimondr) [inline ensure_command_output.sh]: #!/bin/bash -ex

**Comment by arhimondr** [comment]
lgtm % comments

**Comment by cawallin** [comment]
presto-checks build: https://travis-ci.org/Teradata/presto-checks/builds/184668634 (note that the failures are because nothing has been merged to the sprint build, and it's good that there're there because otherwise it wouldn't show that the change is working :) ). Compare it with https://travis-ci.org/Teradata/presto-checks/builds/184668240 without this change.

File: ensure_command_output.sh (hunk 6 above)
Comment (ArturGajowy) [reply ensure_command_output.sh]: I think you meant `-e`, we don't want debug noise from this scirpt

**Review by ArturGajowy — APPROVED** [review]
I think it's the 4th of 5th time when somebody does a change to how product tests are run and suddenly they always pass because of exit code being discarded...

Thanks for catching that!

I think @arhimondr did a great review already. Only one remark from me - please ensure there are no `set -x` debug logs in the output (see my comment @ the shebang).

@fiedukow take a look and add your review if you feel like it :)

**Comment by ArturGajowy** [comment]
Also, congrats on the line count: +20 -40 :)

**Comment by cawallin** [comment]
@fiedukow mentioned that the pipes were intentional, because they ensure that output is synchronized. He said that just adding "-e" to #/bin/bash should fix the script, I'm trying it out now.

**Comment by arhimondr** [comment]
@fiedukow Why do we need to have it sychronized? Do you suspect it can get buffed and not printed out?

**Comment by cawallin** [comment]
I merged this, we can talk about buffering/non-buffering later

**Pull request merged**